src/protocol.jl

```diff
@@ -1,9 +1,9 @@
 # Position
 type Position
-    line::Int
-    character::Int
+    line::Integer
+    character::Integer
```

Comment:
We should use concrete leave types here, i.e. in type declarations, otherwise these values will always be boxed, which is inefficient. I think ``Integer`` is the correct type in the message signature (so that we dispatch correctly), but not in type declarations. I would go with ``Int`` here, that should be faster on x86, and should be plenty in terms of domain.